src/preferences/dialog/dlgprefsound.cpp

```diff
@@ -608,7 +608,8 @@ void DlgPrefSound::updateAudioBufferSizes(int sampleRateIndex) {
                 static_cast<unsigned int>(SoundManagerConfig::
                                 JackAudioBufferSizeIndex::Size4096fpp));
     } else {
-        double sampleRate = sampleRateComboBox->itemData(sampleRateIndex).toDouble();
+        double sampleRate = sampleRateComboBox->itemData(sampleRateIndex)
+                                    .value<mixxx::audio::SampleRate>();
```

Comment:
lets add a debug assert to avoid the issue with the default construction in the future.
```suggestion
        DEBUG_ASSERT(sampleRateComboBox->itemData(sampleRateIndex).canConvert<mixxx::audio::SampleRate>());
        double sampleRate = sampleRateComboBox->itemData(sampleRateIndex)
                                    .value<mixxx::audio::SampleRate>().toDouble();
```